### DISCUSSION TITLE
Windows Terminal v1.15.252

### DISCUSSION THREAD

**Discussion by DHowett** [microsoft/terminal#13976] [Announcements]
This release migrates some awesome features, changes and bug fixes from Terminal 1.15 Preview into the stable channel!

* Terminal now supports "Mark Mode", a keyboard-first text selection and navigation mode. The name is an homage to the traditional Windows Console Host!
    * It is bound by default to <kbd>Ctrl+Shift+M</kbd>

Please see the following release notes for additional details:

* [Windows Terminal Preview v1.15.228](https://github.com/microsoft/terminal/releases/tag/v1.15.2282.0)
* [Windows Terminal Preview v1.15.200](https://github.com/microsoft/terminal/releases/tag/v1.15.2002.0)
* [Windows Terminal Preview v1.15.186](https://github.com/microsoft/terminal/releases/tag/v1.15.1862.0)

Note that the new text rendering engine and scrollbar mark feature is not included in this Stable build. Yet.

> **IMPORTANT**
> This version was made available to the Dev External flighting ring (Windows Insiders) first, and will be
> released to general availability one or two weeks later depending on its reliability.

As a reminder, Terminal 1.12 was the last version of Windows Terminal that supports Windows 19H1 or 19H2.
That version of windows is going out of support soon, so you may want to consider upgrading.

<details>
<summary>Preinstallation Kit info</summary>

A [preinstallation](https://docs.microsoft.com/en-us/windows/msix/desktop/deploy-preinstalled-apps) kit is available for system integrators and OEMs interested in prepackaging Windows Terminal with a Windows image. More information is available in the [DISM documentation on preinstallation](https://docs.microsoft.com/windows-hardware/manufacture/desktop/preinstall-apps-using-dism). Users who do not intend to preinstall Windows Terminal should continue using the _msixbundle_ distribution.

</details>

<details>
<summary>Why are there so many packages? How do I choose?</summary>
This version of Windows Terminal is distributed in two bundles, one of which works on Windows 10-11 and the other of
which only works on Windows 11. The Windows 11 version is much smaller because we no longer need to work around a
platform issue related to our dependencies.

If you intend on using Terminal as an unpackaged application--that is, extracting the `msix` file--we recommend that
you use the `Win10` bundle. You will need the [Visual C++ runtime redistributable](https://docs.microsoft.com/en-us/cpp/windows/latest-supported-vc-redist).

In addition, if you install the _packaged_ version on either Windows 10 or Windows 11, it now depends on [the Visual C++ Universal Runtime Package](https://docs.microsoft.com/en-us/troubleshoot/developer/visualstudio/cpp/libraries/c-runtime-packages-desktop-bridge?msclkid=3e1dd82aac8b11ec8f9dbd86d5979ca8#how-to-install-and-update-desktop-framework-packages).

Despite these distributions having different version numbers, they are built from the same code and there is no
functional difference between them.

If you install the Windows 10 version on Windows 11, it will probably automatically upgrade itself to the Windows 11
version. It turns out that it is impossible to have two bundles with _the same version number_, so it has to be this
way.
</details>

In addition to the above, we've backported the following changes and bugfixes from Windows Terminal Preview 1.16:

# Changes

## Interaction

* When in mark mode, its built in key bindings <kbd>Ctrl+A</kbd> and the modified arrow keys will take precedence over your key bindings (#13659)
* We've polished how existing selections interact with mark mode (#13893)

## UI

* @dansmor7 figured out that we don't need to draw our caption buttons ourselves; now they look great on all versions of Windows! (#13341) (thanks!)

# Bug Fixes

## Interaction

* Terminal will now use the tab's active title for `Export Text` (#13915) (thanks @serd2011!)
* The Emoji picker, PinYin IME or any other IME will no longer drift off the bottom of the screen (oops) (#13785)

## Accessibility

* The Command Palette has become much chattier, announcing (to a screen reader) the name of the selected item (#13519)
* Asking for `INT_MAX` characters via UIA will no longer wig us out or try to send you multiple gigabytes of null bytes (#13779)
    * However, it remains impolite to ask for `INT_MAX` characters via `ITextPattern::GetText`.

## Performance

* Terminal is now 1.2 megabytes smaller on disk (uncompressed) thanks to not using RTTI (#13947) (thanks RTTI!)

## Reliability

* Fixed a number of crashes (smaller number than that in Preview), not all of which were common or user-impacting:
    * Attempted a fix for the `SignalTextChanged` crash (#13876)
    * Attempted another fix, this time for the `_refreshSizeUnderLock` crash (#13857)
    * Fixed a crash in `_WritePseudoWindowCallback` (#13777)
    * Fixed a crash on exit with the command palette open (#13778)
    * Fixed a race condition in UpdatePatternLocations (#13859)
    * Fixed two race conditions around pseudo window visibility (#13832)
    * Fixed a crash in NVDA, caused by us considering a specific text range invalid (#13907)
* Terminal should now more reliably appear in the context menu

_With additional thanks to our documentation and code health contributors @jsoref and @LitoMore._

<hr /><em>This discussion was created from the release <a href='https://github.com/microsoft/terminal/releases/tag/v1.15.2524.0'>Windows Terminal v1.15.252</a>.</em>

**Comment by RoyiAvital** [comment]
@DHowett , Will you add the option to set it as the default shell in Windows 10 like in Windows 11?

**Reply by zadjii-msft** [reply]
Please [read the FAQs](https://github.com/microsoft/terminal/wiki/Frequently-Asked-Questions-(FAQ)#will-the-default-terminal-functionality-be-coming-to-windows-10).
